docs/oauth2.md

```diff
@@ -62,149 +71,81 @@ auth_backends.1 = rabbit_auth_backend_oauth2
-2cr3wQwCfF1qVu4eAVNVfxfy/uEvG3Q7x005P3TcK+QcYgJxav3lictSi5dyWLgG
-QAvkknWitpRK8KVLypEj5WKej6CF8nq30utn15FQg0JkHoqzwiCqqeen8GIPteI7
-VwIDAQAB
------END PUBLIC KEY-----">>}
-          }}
-      ]}
-    ]}
-].
-```
+Tokens must be digitally signed in order to be accepted. RabbitMQ must have the signing key to validate the signature. You can either configure the signing keys the OAuth 2.0 provider will use or configure RabbitMQ with one of the two endpoints below:
 
-If a symmetric key is used, the configuration will look like this:
+  - **JWKS endpoint** - this is the HTTP endpoint that returns the signing keys used to digitally sign the tokens.
+  - **OpenID Provider Configuration endpoint** - this is the endpoint that returns the provider's configuration including all its endpoints, such as the **JWKS endpoint**.
 
-```erlang
-[
-  {rabbitmq_auth_backend_oauth2, [
-    {resource_server_id, <<"my_rabbit_server">>},
-    {key_config, [
-      {signing_keys, #{
-        <<"a-key-ID">> => {map, #{<<"kty">> => <<"MAC">>,
-                                  <<"alg">> => <<"HS256">>,
-                                  <<"value">> => <<"my_signing_key">>}}
-      }}
-    ]}
-  ]},
-].
-```
+When you configure RabbitMQ with one of two aforementioned endpoints, RabbitMQ must make a HTTP request (or two, if we specify the latter endpoint) to download the signing keys. This is an operation that occurs once for any signing key not downloaded yet. When the OAuth 2.0 provider rotates the signing keys, newer tokens refer to a new signing key which RabbitMQ does not have yet which triggers another download of the newer signing keys.
 
-The key set can also be retrieved dynamically from a URL serving a [JWK Set](https://tools.ietf.org/html/rfc7517#section-5).
-In that case, the configuration will look like this:
+The token can be any [JWT token](https://jwt.io/introduction/) which contains the `scope` and `aud` fields.
 
-```erlang
-[
-  {rabbitmq_auth_backend_oauth2, [
-    {resource_server_id, <<"my_rabbit_server">>},
-    {key_config, [
-      {jwks_url, <<"https://my-jwt-issuer/jwks.json">>}
-    ]}
-  ]},
-].
-```
+In chronological order, here is the sequence of events that occur when a client application wants to connect to one of the RabbitMQ's messaging protocols, such as AMQP:
 
-NOTE: `jwks_url` takes precedence over `signing_keys` if both are provided.
+1. Client application requests an **access_token** from the OAuth 2.0 provider.
+2. The **access token** must include in the `scope` field **scopes** supported by RabbitMQ (it is possible to use a different field for the **scopes** by setting the name of the new field in `auth_oauth2.additional_scopes_key`).
+3. Client application passes the token as password when connecting to RabbitMQ's messaging protocol. **The username field is ignored**.
+4. RabbitMQ validates the token's signature. To do so, it must have the signing keys or download them from the JWKS endpoint as explained in earlier sections.
```

Comment:
Change to "To do so" to "To validate it, RabbitMQ must have the signing keys or download them from the JWKS endpoint as explained in earlier sections."

Question I know we mention downloading the signing keys in earlier sections? Is it obvious how to do this for users? we don't document the "downloading" ...just looking back at previous sections - is it automatically triggered?